jet_tracking/context.py

```diff
@@ -1,20 +1,90 @@
 import json
 import logging
 import os
-import threading
 from argparse import ArgumentParser
 from pathlib import Path
 
 import numpy as np
 import yaml
 
-log = logging.getLogger(__name__)
-lock = threading.Lock()
+log = logging.getLogger("jet_tracker")
 
 
 class Context(object):
-
+    """
+    The Context class represents the context or environment in which the application operates.
+    It stores various settings, configurations, and data related to the application's functionality.
+
+    Attributes:
+        signals (object): Object containing different signals emitted by the context.
+        JT_LOC (str): Jet tracking location path.
```

Comment:
Small tip: you can also include these as type hint annotations (outside of the docstring), which will make them show up in IDEs.
```
signals: object
JT_LOC: str
SD_LOD: str
```